builtin/providers/aws/resource_aws_rds_cluster.go

```diff
@@ -294,6 +304,38 @@ func resourceAwsRDSClusterCreate(d *schema.ResourceData, meta interface{}) error
 				return err
 			}
 		}
+
+		if clusterUpdate {
+			log.Printf("[INFO] RDS Cluster is restoring from snapshot with default db_cluster_parameter_group_name and backup_retention_period, " +
+				"but custom values should be set, will now update after snapshot is restored!")
+
+			d.SetId(d.Get("cluster_identifier").(string))
+
+			log.Printf("[INFO] RDS Cluster ID: %s", d.Id())
+
+			log.Println("[INFO] Waiting for RDS Cluster to be available")
+
+			stateConf := &resource.StateChangeConf{
+				Pending:    []string{"creating", "backing-up", "modifying"},
+				Target:     []string{"available"},
+				Refresh:    resourceAwsRDSClusterStateRefreshFunc(d, meta),
+				Timeout:    120 * time.Minute,
+				MinTimeout: 3 * time.Second,
+				Delay:      30 * time.Second, // Wait 30 secs before starting
+			}
+
+			// Wait, catching any errors
+			_, err = stateConf.WaitForState()
+			if err != nil {
+				return err
+			}
+
+			err = resourceAwsRDSClusterUpdate(d, meta)
+			if err != nil {
+				return err
+			}
```

Comment:
Hey there – this `WaitForState` block looks identical to the one at the end there, is that true? If so, could we just use that one? I'm wondering if instead of this block we could just have this:

```
if clusterUpdate {
  log.Printf("[INFO] RDS Cluster is restoring from snapshot....")
  return resourceAwsRDSClusterUpdate(d, meta)
} else {
  return resourceAwsRDSClusterRead(d, meta)
}
```